### PR TITLE
controller refactoring

### PR DESCRIPTION
**Pull request by mozafrank**
The goal of this PR was to clarify the purpose of the controller; historically it has served multiple purposes but with these changes the controller now starts and maintains various threads, and all the work of receiving events and correlating that to checks is done by worker threads which the controller manages. 

This doesn't change the functionality a great deal, but does allow us to add better thread management to the controller, so now it actively checks if all its threads are working and restarts them if not.  It should also speed things up, since the work of 1 thread is now done by 3. This covers https://jira.mozilla.com/browse/SE-1202

Tl;dr:
* Reorganizing code in checks/ and events/
* Move k8s event handling out of the controller to its own thread, the event-handler
* Move check thread management out of the controller to its own thread, the check-handler
* Black formatting

Testing:
* Unit tests have been updated and run
* tested in afrank-dev with limited check load
* tested in mozilla-it-service-engineering under full check load with metrics enabled.

@bowlofstew a review please

### CODE REVIEW

File: mozalert/events/handler.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+from mozalert import kubeclient
+
+
+class EventHandler(threading.Thread):
+    def __init__(self, **kwargs):
+        super().__init__()
+        self.kube = kwargs.get("kube", kubeclient.KubeClient())
+        self.shutdown = kwargs.get("shutdown", lambda: False)
+
+        self.domain = kwargs.get("domain", "crd.k8s.afrank.local")
```
Comment (claudiouzelac) [inline mozalert/events/handler.py:14]: Acceptable default? :)

File: mozalert/controller.py [hunk 2]
```diff
@@ -19,145 +23,134 @@ def __init__(self, **kwargs):
...
-                 check object.
-
-        MODIFIED: this can be triggered by the user patching their check, or by a check thread
-                  updating the object status. NOTE: updating the status subresource SHOULD NOT
-                  trigger a modify, this is probably a bug in k8s. when a check is updated the
-                  changes are applied to the check object.
-
-        ERROR: this can occur sometimes when the CRD is changed; it causes the process to die
-               and restart.
+        the controller runs various threads:
```
Comment (claudiouzelac) [inline mozalert/controller.py:92]: Thank you for the documentation!

**Comment by claudiouzelac** [comment]
I applaud your fantastic pull request description, your attention to detail, and software craftsmen-ship.

**Review by claudiouzelac — APPROVED** [review]
LGTM
